src/main/scala/is/hail/methods/VEP.scala

```diff
@@ -308,74 +311,119 @@ object VEP {
         it
           .grouped(localBlockSize)
           .flatMap { block =>
-            val (jt, proc) = block.iterator.map { case (v, (va, gs)) => v }.pipe(pb,
+            val (jt, proc) = block.iterator.map { rv =>
+              
+              val vOffset = localRowType.loadField(rv, 1)
+              
+              Variant.fromRegionValue(rv.region, vOffset) }
+              .pipe(pb,
               printContext,
               printElement,
               _ => ())
 
-            val nonStarToOriginalVariant = block.map { case (v, (va, gs)) =>
+            val nonStarToOriginalVariant = block.map { rv =>
+              val vOffset = localRowType.loadField(rv, 1)
+              val v = Variant.fromRegionValue(rv.region, vOffset)
+              
               (v.copy(altAlleles = v.altAlleles.filter(_.alt != "*")), v)
             }.toMap
-
-            val kt = jt
-              .filter(s => !s.isEmpty && s(0) != '#')
-              .map { s =>
+            
+            val rvb = new RegionValueBuilder()
+            val rv2 = RegionValue()
+            
+            val t: TStruct =
+              if (csq)
+                TStruct("v" -> localVariantType, root -> TArray(TString()))
+              else 
+                TStruct("v" -> localVariantType, root -> vepSignature)
+            
+            val kt = block.map(_.region).iterator.zip(jt) // FIXME: rather than zip, should i create a new memory buffer?
```

Comment:
Slightly revising what I said earlier: when you get the results back for the block, you should pull out the variants with `variantFromInput`, sort that, then return an iterator of region values which converts the results for each variant one at a time.  It doesn't really matter when/where this conversation happens.  Best would import it directly into the region value where the final result is being built inside inserter given to `insertIntoRow`.